pkg/controller/controller.go

```diff
@@ -34,6 +35,20 @@ type Reconciler struct {
 // handleTaskRun handles a changed or created TaskRun.
 // This is the main entrypoint for chains business logic.
 func (r *Reconciler) handleTaskRun(ctx context.Context, tr *v1beta1.TaskRun) error {
+	// Check to make sure the TaskRun is finished.
+	if !tr.IsDone() {
+		r.Logger.Infof("taskrun %s/%s is still running", tr.Namespace, tr.Name)
+		return nil
+	}
+	// Check to see if it has already been signed.
+	if signing.IsSigned(tr) {
```

Comment:
Does Tekton have a concept of an immutable `TaskRun`? Since users could still modify the TaskRun to manipulate this annotation after it has been added and after they modify the TaskRun, should we concern ourselves with making sure this is immutable?

Reply:
Tekton/Kubernetes do not really allow immutability to be enforced. I think we'll need to rely on RBAC here.

Reply:
I'm not really sure it's worth investigating at this point, but I'm thinking of the use of admission controllers to enforce immutability for certain TaskRuns that require it e.g. ones that have already been signed.